parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -539,7 +540,14 @@ private static final ParquetMetadata readFooter(InputFile file, ParquetReadOptio
       throw new RuntimeException("corrupted file: the footer index is not within the file: " + footerIndex);
     }
     f.seek(footerIndex);
-    return converter.readParquetMetadata(f, options.getMetadataFilter());
+    // Read all the footer bytes in one time to avoid multiple read operations,
+    // since it can be pretty time consuming for a single read operation in HDFS.
+    ByteBuffer footerBytesBuffer = ByteBuffer.allocate(footerLength);
+    f.readFully(footerBytesBuffer);
+    LOG.debug("Finished to read all footer bytes.");
+    footerBytesBuffer.flip();
+    InputStream footerBytesStream = ByteBufferInputStream.wrap(footerBytesBuffer);
```

Comment:
Shouldn't we close this input stream?

Reply:
PTAL: https://github.com/apache/parquet-mr/pull/685/

Reply:
Technically, you're right. But `ByteBufferInputStream.close()` does nothing. So it is not a bug to not close it.